templates/fine-tune-function-calling/README.ipynb

```diff
@@ -0,0 +1,761 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get your API key from https://console.anyscale.com/credentials\n",
+    "ANYSCALE_API_KEY = \"esecret_yourKeyHere\"  \n",
+    "ANYSCALE_API_BASE = \"https://api.endpoints.anyscale.com/v1\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Anyscale Endpoints are OpenAI compatible\n",
+    "client = openai.OpenAI(\n",
+    "    base_url = ANYSCALE_API_BASE,\n",
+    "    api_key = ANYSCALE_API_KEY\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Upload the files to Anyscale\n",
+    "training_file_id = client.files.create(\n",
+    "    file=open(train_file_path,'rb'),\n",
+    "    purpose=\"fine-tune\",\n",
+    ").id\n",
+    "\n",
+    "valid_file_id = client.files.create(\n",
+    "    file=open(validation_file_path,'rb'),\n",
+    "    purpose=\"fine-tune\",\n",
+    ").id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's now launch a fine-tuning job for 4 epochs. The expected time for this job is < 3 hours. For instructions on viewing job status, other hyperparameters used, etc, you can refer to our [fine-tuning guide](https://docs.anyscale.com/preview/examples/e2e-finetune-and-serve-example#4-start-the-fine-tuning). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create finetuning job. Other parameters like context length will be chosen appropriately based on dataset size\n",
+    "fine_tuning_job_id = client.fine_tuning.jobs.create(\n",
+    "    model=\"meta-llama/Meta-Llama-3-8B-Instruct\",\n",
+    "    hyperparameters={\"n_epochs\": 4},\n",
+    "    training_file=training_file_id,\n",
+    "    validation_file=valid_file_id,\n",
+    ").id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 3: Serving\n",
+    "\n",
+    "## Step 3(a): Finetuned on the Anyscale Platform\n",
+    "\n",
+    "Make a note of the final checkpoint after fine-tuning (this should be the last line in the logs). You can now spin up the \"Deploy LLMs\" template which has all the instructions and required dependencies to serve your finetuned model efficiently. You will find the tutorials on [serving LoRA models](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/lora/DeployLora.ipynb) (if applicable) and on deploying a [custom model](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/CustomModels.ipynb) helpful. Once you have set up your fine-tuned model as an Anyscale Service, head over to the \"Services\" tab in the console and select your deployed service. \n",
```

Comment:
Can you update this based on the new flow in the template with a generate config script?

Reply:
Hmm this won't be a standalone template so just I'm directing them to the serve template - that will have all the instructions. So wouldn't it be better to keep this as is?

Reply:
+1 to @SumanthRH